executor/runcexecutor/executor.go

```diff
@@ -524,14 +532,50 @@ func (p *procHandle) WaitForReady(ctx context.Context) error {
 func (p *procHandle) WaitForStart(ctx context.Context, startedCh <-chan int, started func()) error {
 	startedCtx, timeout := context.WithTimeout(ctx, 10*time.Second)
 	defer timeout()
-	var err error
 	select {
 	case <-startedCtx.Done():
 		return errors.New("runc started message never received")
-	case pid, ok := <-startedCh:
+	case runcPid, ok := <-startedCh:
 		if !ok {
 			return errors.New("runc process failed to send pid")
 		}
+
+		// we need the process for the runc monitor process so we can send
+		// SIGWINCH signals to it, if we send them to the process inside the
+		// container nothing happens.
+		var err error
+		p.MonitorProcess, err = os.FindProcess(runcPid)
+		if err != nil {
+			return errors.Wrapf(err, "unable to find runc monitor process for pid %d", runcPid)
+		}
```

Comment:
if this code is linux-specific you can safely do just `p.MonitorProcess, _ = os.FindProcess(runcPid)`. See below for an explanation.